src/android/com/megster/cordova/BluetoothSerial.java

```diff
@@ -354,14 +355,75 @@ public boolean execute(String action, CordovaArgs args, CallbackContext callback
         return validAction;
     }
 
+    public void unPairDevice(CordovaArgs args, CallbackContext callbackContext) {
+        String macAddress = null;
+        try {
+            macAddress = args.getString(0);
+
+            BluetoothDevice device = bluetoothAdapter.getRemoteDevice(macAddress);
+            Log.d(TAG, "## Processing incoming connect for device " + device.getName());
+            if (device != null) {
+                try {
+                    Method method = device.getClass().getMethod("removeBond", (Class[]) null);
+                    method.invoke(device, (Object[]) null);
+                    LOG.i(TAG, "Successfully removed bond");
+                } catch (Exception e) {
+                    LOG.e(TAG, "ERROR: could not remove bond");
+                    e.printStackTrace();
+                }
+                callbackContext.success();
+            } else {
+                String message = "Peripheral " + macAddress + " not found.";
+                LOG.w(TAG, message);
+                callbackContext.error(message);
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * This enum lists all the classic devices which use Serial Port Profile for connection. It's is one of the more fundamental Bluetooth profiles.
+     * Using SPP, each connected device can send and receive data just as if there were RX and TX lines connected between them.
+     * For us, AnD classic devices use this. For future, add any other device which uses SPP here.
+     */
+    public enum SPPProfileDevice {
```

Comment:
This enum lists all the classic devices which use Serial Port Profile for connection. It's is one of the more fundamental Bluetooth profiles. Using SPP, each connected device can send and receive data just as if there were RX and TX lines connected between them. For us, AnD classic devices use this. For future, add any other device which uses SPP here.